framework/src/media/FileOutputDataSource.cpp

```diff
@@ -88,17 +94,19 @@ bool FileOutputDataSource::open()
 bool FileOutputDataSource::close()
 {
 	switch (getAudioType()) {
-		case AUDIO_TYPE_WAVE:
-			unsigned int fileSize;
-			fileSize = ftell(mFp);
-			if (fileSize < 0) {
+		case AUDIO_TYPE_WAVE: {
+			fflush(mFp);
+			long ret = ftell(mFp);
+			if (ret < 0) {
```

Comment:
please add more info for error case, 
meddbg("file size could not be found errno : %d\n", errno);

Reply:
Fixed, thanks.